### DISCUSSION TITLE
How to modify the default configuration of a route？

### DISCUSSION THREAD

**Discussion by CrazyPotato424** [openTCS/opentcs#144] [Q&A]
How to modify the default configuration of a route？

**Comment by swltr** [comment]
See https://opentcs.org/docs/5.17/user/opentcs-users-guide.html#_default_router.

**Reply by CrazyPotato424** [reply]
it seems it just can modify the algorithm type and the cost functions but how could i add my own algorithm to the system?

**Reply by swltr** [reply]
@zsjyg Via configuration files, only the documented algorithms can be selected (see `DefaultRouterModule.java`) right now. To add your own algorithms, you'd have to register a Guice module in which you bind your own algorithm implementation via something like `bind(PointRouterFactory.class).to(MyPointRouterFactory.class)`.
